node/pkg/watchers/stacks/fetch.go

```diff
@@ -0,0 +1,305 @@
+		NetworkID              uint64                 `json:"network_id"`
+		ParentNetworkID        uint64                 `json:"parent_network_id"`
+		StacksTipHeight        uint64                 `json:"stacks_tip_height"`
+		StacksTip              string                 `json:"stacks_tip"`
+		StacksTipConsensusHash string                 `json:"stacks_tip_consensus_hash"`
+		GenesisChainStateHash  string                 `json:"genesis_chainstate_hash"`
+		UnanchoredTip          *string                `json:"unanchored_tip"`
+		UnanchoredSeq          *uint64                `json:"unanchored_seq"`
+		TenureHeight           uint64                 `json:"tenure_height"`
+		ExitAtBlockHeight      *uint64                `json:"exit_at_block_height"`
+		IsFullySynced          bool                   `json:"is_fully_synced"`
+		NodePublicKey          string                 `json:"node_public_key"`
+		NodePublicKeyHash      string                 `json:"node_public_key_hash"`
+		LastPoxAnchor          *StacksV2InfoPoxAnchor `json:"last_pox_anchor"`
+		Stackerdbs             []string               `json:"stackerdbs"`
+	}
+)
+
+// Fetches a tenure and its blocks by Bitcoin (burn) block height
+func (w *Watcher) fetchTenureBlocksByBurnHeight(ctx context.Context, height uint64) (*StacksV3TenureBlocksResponse, error) {
+	url := fmt.Sprintf("%s/v3/tenures/blocks/height/%d", w.rpcURL, height)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch Bitcoin (burn) block: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		// Burn block was skipped, return empty tenure blocks
+		return &StacksV3TenureBlocksResponse{
+			BurnBlockHeight: height,
+			StacksBlocks:    []StacksV3TenureBlock{},
+		}, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("unexpected status code: %d", resp.StatusCode)
+	}
+
+	body, err := common.SafeRead(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
+	var tenureBlocks StacksV3TenureBlocksResponse
+	if err := json.Unmarshal(body, &tenureBlocks); err != nil {
+		return nil, fmt.Errorf("failed to parse response: %w", err)
+	}
+
+	return &tenureBlocks, nil
+}
+
+// Fetches block replay data including all transactions for a given block
+// Uses the v3 blocks/replay endpoint which includes vm_error for failed transactions
+func (w *Watcher) fetchStacksBlockReplay(ctx context.Context, blockId string) (*StacksV3TenureBlockReplayResponse, error) {
+	url := fmt.Sprintf("%s/v3/blocks/replay/%s", w.rpcURL, blockId)
+
+	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create request: %w", err)
+	}
+
+	if w.rpcAuthToken != "" {
+		req.Header.Set("Authorization", w.rpcAuthToken)
```

Comment:
Can you clarify which of these endpoints need authentication and why? 
It may be better to create an http client in the Watcher object itself and configure it with an API key rather than re-implementing a DefaultClient in each of these methods.